### ISSUE TITLE
Assignment 6: Non-rigid Alignment 

### ISSUE THREAD

**Issue by Chiaki530** [eth-igl/GP2021-Assignments#87]
I have some questions about the non-rigid alignment process (as shown in slides 11-12) that I couldn't understand. 
(1) To implement the iterative optimization for the warping energy function, does it mean to perform gradient descent with a fixed learning rate to update x' iteratively? 
(2) Should we treat the Lagrange multiplier lambda as a predefined constant or as another variable that we can update? 
(3) Why can't we use a similar substitution method in assignment 5 for this problem?

**Comment by wsh312** [comment]
Hi, in the open project we don't define the master solution, the approach in the slides is just a recommendation based on the implementation and solutions that came up in the past. The task is to understand the problem and define the suitable optimization.  (1) An iterative optimization is more stable since our initialization (after rigid deformation) is still quite far away from the scanned mesh, so gradually adding the constraints (e.g, first only the landmarks, then adding nearby vertexes of the two meshes) could make the optimization more stable and easier to find suitable parameters. (2) I would start by using a constant lambda and see if alternatively optimizing it would help. (3) Yes, you are welcome to start from the solution you had in the assignment (5), e.g., the landmarks are the moved handles, and go from there.

**Comment by wsh312** [comment]
The idea of this project and the bonus task is to allow you to be creative and be rewarded by trying interesting approaches. Ideally, every group comes up with its own method to tackle the problem. So if you tried something but doesn't work well, you are still welcome to put it in the presentation and share your thoughts with us. It would be a bit boring if every group has the same presentation flow in the end ;) But I also suggest starting from the baseline method we suggested before testing a more sophisticated idea.

**Comment by Chiaki530** [comment]
Thanks!
